src/redis-cli.c

```diff
@@ -5864,6 +5865,26 @@ static int clusterManagerCommandAddNode(int argc, char **argv) {
     listAddNodeTail(cluster_manager.nodes, new_node);
     added = 1;
 
+    if (!master_node) {
```

Comment:
@madolson can you review this code, and also maybe advice how to test it?
seems that the cluster tests don't use `redis-cli --cluster add-node`.

Reply:
Yeah, since all of the cluster tests rely a fixed 20 nodes, it's not surprising that it wasn't tested. It's probably easier to create a test in the old framework that starts up a small cluster, and adds in a new node.

Reply:
by "old" you mean `tests/unit/cluster.tcl` that i recently created? or `tests/cluster/tests`?

Reply:
Not sure how to proceed here, should I add tests for adding node to the cluster? Maybe it should be a separate PR?

Reply:
@MeirShpilraien we discussed this in the core-team meeting, and concluded that we'd like it tested as part of `tests/unit/cluster.tcl`

Reply:
@oranagra added a test, let me know what you think.